app/models/article.rb

```diff
@@ -30,17 +32,18 @@ class Article < ActiveRecord::Base
   has_many :articles_courses, class_name: ArticlesCourses
   has_many :courses, -> { uniq }, through: :articles_courses
   has_many :assignments
+  belongs_to :wiki
+
+  # Allows us to directly #update from the Tools response
+  alias_attribute :page_id, :native_id
 
   scope :live, -> { where(deleted: false) }
   scope :current, -> { joins(:courses).merge(Course.current).uniq }
   scope :namespace, -> ns { where(namespace: ns) }
 
-  # Always save titles with underscores instead of spaces, since that's the way
-  # they are in the MediaWiki database.
   validates :title, presence: true
-  before_validation do
-    self.title = title.tr(' ', '_')
-  end
+
+  before_validation :set_defaults_and_normalize
```

Comment:
Why is before_validate the place to put transitional data into the new fields? Shouldn't we override getters instead?